qiskit/circuit/instruction.py

```diff
@@ -204,6 +204,10 @@ def params(self, parameters):
                 raise CircuitError("invalid param type {0} in instruction "
                                    "{1}".format(type(single_param), self.name))
 
+    def is_parameterized(self):
```

Comment:
What do you think about making this a `@property`?

Reply:
Hard to evaluate that suggestion at my level of Python sophistication :)

Do you mean that as something that would be set `True` when a `Parameter` or `ParameterExpression` is added to the `Instruction`?

Reply:
@kdk I just read up on `@property` ... I guess that would _work_ ... but to my taste as a "software generalist" rather than a "Python aficionado" a "property" is something that other packages both "set" and "get". In the present instance, no caller has the right to "set" `is_parameterized`, which is an opaque indicator that a caller has added a `Parameter` to an `Instruction`.

Reply:
:+1: Adding a `@property` without a corresponding `@is_parameterized.setter` would leave `is_parameterized` as a getter-only property, which is not an uncommon pattern. That said, the circuits API already isn't internally consistent on this topic, so leaving it as-is is okay.